app/src/main/java/com/itsaky/androidide/views/editor/CodeEditorView.java

```diff
@@ -176,23 +176,23 @@ public boolean save() {
             return false;
         }
 
-        final var path = getFile().toPath();
-        if (!isModified() && Files.exists(path)) {
+        final var file = getFile();
+        if (!isModified() && file.exists()) {
             LOG.info(getFile().getName());
             LOG.info("File was not modified. Skipping save operation.");
             return false;
         }
 
-        final var lines = getLines(getEditor().getText());
-        try {
+        final var text = getEditor().getText().toString();
+
+        boolean result = FileIOUtils.writeFileFromString(file, text);
```

Comment:
Could you explain what are the benefits of using `FileIOUtils` here? The "extra line" issue can be fixed with other methods too.

Reply:
Why then add libraries and not use their capabilities? Yes, and this method under the hood has all the necessary checks. I do not really like to write the same type of code when it is already implemented.

Reply:
What about error messages? How the user is supposed to know what was the issue when saving the file. If we just print the error message without any detailed error messages, how do you think we would debug issues? Sometimes, issues are just not reproducible on other devices. What would we do in this situation?

I agree that we should use libraries that we add to the project, but that's simply not possible here. If you look at the previous versions of this file, it used to use the same `FileIOUtils` to write files.

Reply:
Fixing the extra "new line" is possible with Java NIO APIs too. For example, in the `getLines` method, you could check if the last line is blank or not. If it is, remove that line from the list of lines.